### DISCUSSION TITLE
nq (number of search vector per search request)  Limit

### DISCUSSION THREAD

**Discussion by Yashwanth1712094** [milvus-io/milvus#33552] [Q&A and General discussion]
milvus-version ---> 2.2.13
Trying to do a bulk search operation,facing the following error
raise MilvusException(response.status.error_code, response.status.reason)
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=nq [16646] is invalid, nq (number of search vector per search request) should be in range [1, 16384], but got 16646)>

How to increae th nq (max no of vector per search request) limit??

**Comment by yhmo** [comment]
In milvus, the max number of target vectors is 16384, the max topk is 16384.

```
target_vectors = [V1, V2, V3 ...... Vn]
res = collection.search(data=target_vectors,
        .......
)
```

The size of target_vectors must be less than 16384.
If the size exceeds 16384, you need to split one query to two queries. Not possible to increase the limit.
```
target_vectors_1 = [V1, V2, V3 ...... V8000]
res1 = collection.search(data=target_vectors_1,
        .......
)

target_vectors_2 = [V8000, V8002, V8003 ...... Vn]
res2 = collection.search(data=target_vectors_2,
        .......
)
```
